tox.ini

```diff
@@ -1,32 +1,90 @@
-    python src/MCPClient/lib/ensure_no_mutable_globals.py
+    # General
+    LOGNAME = user
+    DJANGO_SETTINGS_MODULE = {env:DJANGO_SETTINGS_MODULE:settings.test}
+    PYTEST_ADDOPTS = {env:PYTEST_ADDOPTS:}
+    SRC_DIR = {toxinidir}/src
+    HACK_DIR = {toxinidir}/hack
+    # Archivematica Common
+    ARCHIVEMATICA_COMMON_ROOT = {env:SRC_DIR}/archivematicaCommon
+    ARCHIVEMATICA_COMMON_DIR = {env:ARCHIVEMATICA_COMMON_ROOT}/lib
+    # Dashboard
+    DASHBOARD_ROOT = {env:SRC_DIR}/dashboard
+    DASHBOARD_DIR = {env:DASHBOARD_ROOT}/src
+    DASHBOARD_PYTHONPATH = {env:DASHBOARD_DIR}:{env:ARCHIVEMATICA_COMMON_DIR}
+    # MCP Server
+    MCPSERVER_ROOT = {env:SRC_DIR}/MCPServer
+    MCPSERVER_DIR = {env:MCPSERVER_ROOT}/lib
+    MCPSERVER_PYTHONPATH = {env:MCPSERVER_DIR}:{env:ARCHIVEMATICA_COMMON_DIR}:{env:DASHBOARD_DIR}
+    # MCP Client
+    MCPCLIENT_ROOT = {env:SRC_DIR}/MCPClient
+    MCPCLIENT_DIR = {env:MCPCLIENT_ROOT}/lib
+    MCPCLIENT_PYTHONPATH = {env:MCPCLIENT_DIR}:{env:ARCHIVEMATICA_COMMON_DIR}:{env:DASHBOARD_DIR}
+    # Storage Service
+    STORAGE_SERVICE_ROOT = {env:HACK_DIR}/submodules/archivematica-storage-service
+    STORAGE_SERVICE_DIR = {env:STORAGE_SERVICE_ROOT}/storage_service
+    STORAGE_SERVICE_PYTHONPATH = {env:STORAGE_SERVICE_DIR}:{env:STORAGE_SERVICE_DIR}/storage_service
+    # TOXENV-specific
+    archivematica-common: PYTHONPATH = {env:DASHBOARD_PYTHONPATH}
+    dashboard: PYTHONPATH = {env:DASHBOARD_PYTHONPATH}
+    mcpserver: PYTHONPATH = {env:MCPSERVER_PYTHONPATH}
+    mcpclient: PYTHONPATH = {env:MCPCLIENT_PYTHONPATH}
+    storage-service: PYTHONPATH = {env:STORAGE_SERVICE_PYTHONPATH}
+    checkformigrations-dashboard: PYTHONPATH = {env:DASHBOARD_PYTHONPATH}
+    checkformigrations-storage-service: PYTHONPATH = {env:STORAGE_SERVICE_PYTHONPATH}
+    # Setting HOME prevents Python's pwd module to ask for a real uid inside
+    # the container, and using {temp_dir} allows caching the pre-commit and flake8
+    # dependencies in the tox host
+    linting: HOME = {temp_dir}/user
+
+[testenv:archivematica-common]
+changedir = {env:ARCHIVEMATICA_COMMON_ROOT}
+
+[testenv:dashboard]
+changedir = {env:DASHBOARD_ROOT}
 
 [testenv:mcpserver]
-skip_install = True
-setenv =
-    DJANGO_SETTINGS_MODULE = settings.test
-    PYTHONPATH = src/archivematicaCommon/lib:src/MCPServer/lib:src/dashboard/src
-commands =
-    pip install -q -r requirements-dev.txt
-    py.test --ignore=src/dashboard --ignore=src/MCPClient --ignore=src/archivematicaCommon {posargs}
-
-[testenv:checkformigrations]
-whitelist_externals = bash
-commands =
-    pip install -q -r requirements-dev.txt
-    bash ./hack/checkformigrations.sh
+changedir = {env:MCPSERVER_ROOT}
+
+[testenv:mcpclient]
+changedir = {env:MCPCLIENT_ROOT}
+
+[testenv:mcpclient-ensure-no-mutable-globals]
+commands = python {env:MCPCLIENT_DIR}/ensure_no_mutable_globals.py
+
+[testenv:storage-service]
+deps =
+    -r{env:STORAGE_SERVICE_ROOT}/requirements/production.txt
```

Comment:
It's because the `settings.testmysql` module of the SS [imports `dj-database-url`](https://github.com/artefactual/archivematica-storage-service/blob/ff51028190b5389f54038ac544d1e4d5600491de/storage_service/storage_service/settings/testmysql.py#L5) which is not included in `requirements/base.in` or `requirements/test.in`.